### PR TITLE
[Data Viz] Separate Chart logic from modal 

### PR DESCRIPTION
**Pull request by ajpal**
# Description
Basically the same as #32071 

Old design:
- DataVisualizer: responsible for opening and closing the modal, parsing records, and rendering bar chart, histogram, and scatter plot
- DropdownField: simple dropdown component to select from a set of options
- CrossTabChart: creates a pivot table from the records and renders it as a HTML table

New design
- VisualizerModal: responsible for opening and closing the modal, parsing records, and keeping track of state of selected chart options
- DropdownField: no change
- DataVisualizer: wrapper for both chart types. Contains logic that is shared for all charts (currently just ignoreMissingValues)
- GoogleChartWrapper: Component that knows how to render a GoogleChart into a containing div.
- CrossTabChart: no change

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/package.json [hunk 1]
```diff
@@ -34,8 +34,7 @@
     "storybook:deploy": "./script/deploy-storybook.sh"
   },
   "resolutions": {
-    "eshost": "github:pcardune/eshost#f3fa7fa",
-    "test262-compiler": "github:pcardune/test262-compiler#ffa0dff"
```
Comment (islemaster) [inline apps/package.json:38]: Why were these existing package resolutions removed?

Comment (ajpal) [reply apps/package.json:38]: Hmm, I'm not exactly sure. I ran `yarn add memoize-one` then `yarn` and this was the result. Should I change it back?

Comment (islemaster) [reply apps/package.json:38]: Huh.. based on [this very old comment](https://github.com/code-dot-org/code-dot-org/pull/13974/files#r108005090), maybe not.  Maybe this was never needed, and the relevant issues that we forked for seem to have been resolved too.

File: apps/src/storage/dataBrowser/dataUtils.js [hunk 2]
```diff
@@ -15,6 +15,22 @@ export function getDatasetInfo(tableName) {
...
+  if (records && columns) {
+    let filteredRecords = records;
+    columns.forEach(column => {
+      filteredRecords = filteredRecords.filter(
+        record => !isBlank(record[column])
+      );
+    });
+    return filteredRecords;
+  }
+}
```
Comment (islemaster) [inline apps/src/storage/dataBrowser/dataUtils.js:39]: :+1:

File: apps/src/storage/dataBrowser/dataUtils.js [hunk 3]
```diff
@@ -15,6 +15,22 @@ export function getDatasetInfo(tableName) {
...
+
+export function ignoreMissingValues(records, columns) {
+  if (records && columns) {
+    let filteredRecords = records;
+    columns.forEach(column => {
+      filteredRecords = filteredRecords.filter(
+        record => !isBlank(record[column])
+      );
+    });
+    return filteredRecords;
```
Comment (islemaster) [inline apps/src/storage/dataBrowser/dataUtils.js]: Nit (although I don't think we've configured our linter to forbid this): We're leaning on the implicit `return undefined` fall-through behavior here, and haven't documented it as such.  It seems like an empty array might be a better default case.

File: apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      records.every(record => isNumericOrBlank(record[column]));
+    return columns.filter(column => isColumnNumeric(records, column));
+  });
+
+  render() {
+    const parsedRecords = this.parseRecords(this.props.tableRecords);
+    const numericColumns = this.findNumericColumns(
+      parsedRecords,
+      this.props.tableColumns
+    );
```
Comment (islemaster) [inline apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx:67]: Nice!  So clean!

**Review by islemaster — APPROVED** [review]
This looks great!  Thanks for the careful iteration on this feature!

**Pull request merged**